shopfloor/actions/search.py

```diff
@@ -28,11 +30,20 @@ def package_from_scan(self, barcode):
             return model.browse()
         return model.search([("name", "=", barcode)], limit=1)
 
-    def picking_from_scan(self, barcode):
+    def picking_from_scan(self, barcode, use_origin=False):
         model = self.env["stock.picking"]
         if not barcode:
             return model.browse()
-        return model.search([("name", "=", barcode)], limit=1)
+        domain = [("name", "=", barcode)]
+        if not use_origin:
+            return model.search(domain, limit=1)
+        source_document_domain = [
+            "&",
+            ("origin", "=", barcode),
+            ("state", "=", "assigned"),
```

Comment:
Why are we checking the state when searching against `origin` and not with `name`? IMO we should not check the `state` of the transfer at all in this method, it's the job of the `_check_picking_status` general method. So we could write:
```python
    def picking_from_scan(self, barcode, use_origin=False):
        model = self.env["stock.picking"]
        if not barcode:
            return model.browse()
        domain = [("name", "=", barcode)]
        if use_origin:
            origin_domain = [("origin", "=", barcode)]
            domain = expression.OR([domain, origin_domain])
        return model.search(domain)
```